controllers/releaseplan/adapter.go

```diff
@@ -87,6 +87,11 @@ func (a *adapter) EnsureOwnerReferenceIsSet() (controller.OperationResult, error
 // EnsureMatchingInformationIsSet is an operation that will ensure that the ReleasePlan has updated matching
 // information in its status.
 func (a *adapter) EnsureMatchingInformationIsSet() (controller.OperationResult, error) {
+	// If the releasePlan has no Target set, no ReleasePlanAdmission will be loaded
```

Comment:
I don't think you need this change. The releasePlanAdmission will be nil on line 91/96, so then we will set it unmatched already

Reply:
I will crosscheck, but I added this here because the controller was panic-ing on `GetMatchingReleasePlanAdmission` as the releasePlanAdmission won't be passed for tenant pipelines.

Reply:
Yeah, please crosscheck because we have an issue if so. We should hit this line https://github.com/konflux-ci/release-service/blob/main/loader/loader.go#L122 with the error being a IsNotFound error so we should return `nil, err` (or even if it returns no err but an empty list on line 119, the loop on line 128 would be skipped and we would hit line 142)

Reply:
Adding it again, turns out the problem was not `panic` (it probably was fixed in another place), but anyhow without this line, I get the following error, and the pipeline is not created:

```
2024-05-21T16:59:55.886Z	ERROR	Reconciler error	{"controller": "release", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Release", "Release": {"name":"tenant-pipeline-c7x4z","namespace":"dev-release-team-tenant"}, "namespace": "dev-release-team-tenant", "name": "tenant-pipeline-c7x4z", "reconcileID": "4d625aed-a649-4df2-8de6-97892e2bbb21", "error": "no ReleasePlanAdmission found in namespace (dev-release-team-tenant) with the origin (dev-release-team-tenant) for application 'tenant-pipeline'"}
```
The options would be keeping the code to continue processing or change the logic below it to save the unmatched information, although I think this would not make much sense for tenant pipelines. WDY?

Reply:
This error is not from this controller. You can see `{"controller": "release", "controllerGroup": 
"appstudio.redhat.com", "controllerKind": "Release",` it isn't the ReleasePlan controller

Reply:
I noticed that. I was wondering if the identification of the logging for any reason could be wrong, because it is intriguing that it only breaks if I don't bypass specifically this block.

Reply:
I will try testing this later and report back

Reply:
For whatever reason, my clusterbot cluster was not firing any controllers when I updated RPs. No idea why. Anyways, I think the issue is https://github.com/konflux-ci/release-service/pull/451/files#diff-8e9af1e483632eac17610237caaa5f879ea7bfb3e1e1ea228e45a17c770b9432R117
That should be `!=` Can you fix that and see if you still get an error when not modifying the releaseplan adapter function?

Reply:
Sorry not replying here before. Just got lost making all changes requested. The code is not the one to be tested anymore. I will move the PR to draft until I get proper version to be reviewed.

Reply:
you can close this one as there is not change in this function anymore.